Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut

```diff
@@ -1117,8 +1122,17 @@ int function ServerSortLogic ( serverStruct a, serverStruct b )
 	bool direction
 
 	// We can hard code this cause adding entire columns isn't as easy
-	switch ( filterDirection.sortingBy )
-	{
+	switch ( filterDirection.sortingBy ) 
+  {
+		case sortingBy.DEFAULT:
+			aTemp = a.serverPlayers
+			bTemp = b.serverPlayers
+			if (aTemp + 1 >= a.serverPlayersMax)
+				aTemp = 0
+			if (bTemp + 1 >= b.serverPlayersMax)
+				bTemp = 0
```

Comment:
How about... ?

```suggestion

			// We sort by playercount but multiply playercount with
			// * 1000^2 for `< serverPlayersMax-1`
			// * 1000   for `serverPlayersMax-1`
			// * leave as is for rest
			//
			// This way we get a sort order of
			//  1. `< serverPlayersMax-1`
			//  2. `serverPlayersMax-1`   (nearly full)
			//  3. `serverPlayersMax`     (full)
			//  4. `0`                    (empty)
			// where the individual sections are also sorted internally

			if (aTemp + 1 < a.serverPlayersMax)
				aTemp = aTemp*1000*1000
			if (bTemp + 1 < b.serverPlayersMax)
				bTemp = bTemp*1000*1000
			if (aTemp + 1 == a.serverPlayersMax)
				aTemp = aTemp*1000
			if (bTemp + 1 == b.serverPlayersMax)
				bTemp = bTemp*1000

```

Reply:
I had come up with a solution for when i made a ping sorting algorithm (where -1 meant unknown ping) and i wanted -1 to be sorted after everything but i can't remember how I did it. Also, adding 1000 and 10000 would be better than multiplying the values cause it's less computationally expensive

Reply:
I tried adding first but that puts empty servers before full ones unless I add edge case handling for them. Multiply allows to keep things smaller/simpler in that regard ^^

Regarding the computational expensiveness, as the operations for both addition and multiplication essentially turn to `ADD` and `MUL` assembly instructions when the respective Squirrel code is interpreted, the difference on a typical [Skylake CPU is in the worst case 4x](https://www.agner.org/optimize/instruction_tables.pdf#page=262). So I'd say negligible considering that the sort is only run when opening the server browser (or reloading the list) :P